### PR TITLE
Add font smoothing by default to the body

### PR DESCRIPTION
**Pull request by kof**
## Description

Adding font smoothing by default to the Body component

https://discord.com/channels/955905230107738152/955907787299360788/1132882679189229649

```
  -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
```

## Steps for reproduction

1. open any project, see in dev tools these 2 added (or check visually)

## Code Review

- [ ] hi @istarkov , I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

**Review by TrySound — APPROVED** [review]

**Comment by taylornowotny** [comment]
Screenshot 2023-07-24 at 9 06 24 AM

Text looks much nicer now! 👍

The new code is showing in the CSS preview for every item on the canvas.

**Comment by kof** [comment]
@taylornowotny I am not sure why we are showing inherited properties in the css preview, should we remove those?

**Comment by kof** [comment]
On the second thought its actually correct to show it there, because once we have CSS editor, user should be able to change those things as well

**Comment by kof** [comment]
I think we could in the future group properties into sections: 
- local
- inherited
- cascaded
so it will be clear where each value is coming from just like it is in the style panel

**Comment by kof** [comment]
For now I can sort the properties, showing first the local values, then inherited, then cascaded from breakpoints

**Comment by taylornowotny** [comment]
Here's my suggestion:

Show only local and cascaded styles.
Do not show inherited styles.
This is how Webflow is doing it. No need for categories.

Reason: This window is starting to be overloaded with content. Removing less relevant information helps the user find the more relevant information more quickly. In this case the inherited styles are already displayed on the parent elements so repeating that information on every child is not necessary.

**Comment by kof** [comment]
@taylornowotny how do you justify showing cascaded styles but not showing inherited? When do you draw the line?

**Comment by taylornowotny** [comment]
> @taylornowotny how do you justify showing cascaded styles but not showing inherited? When do you draw the line?

Inherited styles are on the parent so they are easily accessible without switching breakpoints. Furthermore, repeating inherited styles on every child element is redundant and makes it slower to find the info about local styles.

It would be fine to show local styles only, not cascaded, but if a user wants to see those cascaded styles then they would have to switch breakpoints, which may reset their canvas size, thus potentially disrupting their work flow.

**Comment by kof** [comment]
I think we can fix the problem of "only interested in token styles" by showing them first and making sure its clear these are token styles, I am implementing right now grouping where each group will have an inline comment above.

This way we don't have to decide which styles that apply to an element we show and which we don't, does this work for you?

**Comment by taylornowotny** [comment]
That works, but not as good as removing inherited styles.

It still leaves the CSS Preview window overloaded with content and the order of local -> cascaded -> inherited goes against the direction of the CSS cascade of top -> bottom

**Comment by kof** [comment]
here is how it looks like

Screenshot 2023-07-24 at 20 42 07

**Comment by kof** [comment]
Regarding the order you are correct, cascaded should go first.

I will leave them all in for now. I think we need a better decision later, if we go after the ui devtools have, then they all should be in there, maybe for a cleaner look we can later collapse inherited/cascaded styles

Screenshot 2023-07-24 at 20 46 44

**Comment by taylornowotny** [comment]
Why do we need a better decision later? I think my suggestion was ideal.

With inherited styles gone, you wouldn't even need to label "Style Sources", only "Cascaded".

**Comment by kof** [comment]
> Why do we need a better decision later? I think my suggestion was ideal.

Because I think user may equally want to know what's inherited, whats cascaded and we can't remove just inherited, since this would be inconsistent + we do show all of that info in the style panel, so it would be also inconsistent with the style panel to not show that avlue

File: apps/builder/app/builder/features/sidebar-left/navigator/css-preview.tsx [hunk 1]
```diff
@@ -21,33 +22,62 @@ const preStyle = css(textVariants.mono, {
...
+    }
+    if (value.cascaded?.value) {
+      cascadedStyles.set(property, value.value);
+      continue;
+    }
+    if (value.inherited?.value) {
+      inheritedStyles.set(property, value.value);
+      continue;
+    }
+  }
```
Comment (istarkov) [inline apps/builder/app/builder/features/sidebar-left/navigator/css-preview.tsx:60]: What if we get here? Should be some exchaustive check if something will change in the future

**Comment by kof** [comment]
> We can remove just inherited because the user can find all those styles on the parent anyway.

Without knowing what has been actually inherited - no, because not every property is inheritable. They would have to do gymnastics to understand what are the values the child will possible inherit from the parent.

File: apps/builder/app/builder/features/sidebar-left/navigator/css-preview.tsx (hunk 1 above)
Comment (kof) [reply apps/builder/app/builder/features/sidebar-left/navigator/css-preview.tsx:60]: please explain

**Review by taylornowotny — APPROVED** [review]

File: apps/builder/app/builder/features/sidebar-left/navigator/css-preview.tsx (hunk 1 above)
Comment (istarkov) [reply apps/builder/app/builder/features/sidebar-left/navigator/css-preview.tsx:60]: You have a huge series of if's
In some places you explicitly do nothing like 
```
 if (value === undefined) {
      continue;
    }
```
At the end of series you or should cover all possibilities, and have check that code will not be occured there.
For example we will add additional style etc 
Example https://github.com/webstudio-is/webstudio-builder/blob/61f164d40e6c62bc98a0f9f5cb894984b2d93bd5/packages/css-engine/src/core/to-value.ts#L7

Comment (istarkov) [reply apps/builder/app/builder/features/sidebar-left/navigator/css-preview.tsx:60]: or give a comment that other possibilities exists and it's not an error that you are doing nothing

Comment (kof) [reply apps/builder/app/builder/features/sidebar-left/navigator/css-preview.tsx:60]: I am confused, ofc its its not an error, we are exiting early beause the value was handled, its a typical early return situation though with a continue ... I don't know what can be misunderstood

Comment (istarkov) [reply apps/builder/app/builder/features/sidebar-left/navigator/css-preview.tsx:60]: Its not misunderstanding its error.
See example I gave with `assertUnreachable` every time we add new style type my code give ts error.
Your code if we will add additional source will never somehow alert us that new style source exists and must be fixed here

**Review by istarkov — APPROVED** [review]

**Comment by kof** [comment]
@TrySound @istarkov I found a way to create a Utility function that does all 3 things for our error handling:
- throws in development runtime
- throws in typescript
- doesn't break production, but lets sentry catch it in the future and logs the error

File: packages/error-utils/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# CSS Vars utils
+
+This package helps you create uniquely named css variables as well as use them.
+
+## API
+
+```ts
+import { cssVars } from "error-utils";
```
Comment (TrySound) [inline packages/error-utils/README.md:8]: Wrong docs?

**Review by istarkov — APPROVED** [review]

**Review by TrySound — APPROVED** [review]

File: packages/error-utils/README.md (hunk 2 above)
Comment (kof) [reply packages/error-utils/README.md:8]: yep

**Pull request merged**